src/scripts/main.js

```diff
@@ -1 +1,94 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      clearTimeout(timeoutId);
+      resolve('First promise was resolved');
```

Comment:
Consider removing the click event listener after resolving the promise to prevent potential memory leaks.